api/types/sortoptions.go

```diff
@@ -0,0 +1,28 @@
+// Copyright 2021 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package types
+
+// SortOptions defines the order that kustomize outputs resources.
+type SortOptions struct {
+	// Order selects the ordering strategy.
+	Order SortOrder `json:"order,omitempty" yaml:"order,omitempty"`
+	// LegacySortOptions tweaks the sorting for the "legacy" sort ordering
+	// strategy.
+	LegacySortOptions *LegacySortOptions `json:"legacySortOptions,omitempty" yaml:"legacySortOptions,omitempty"`
+}
+
+// SortOrder defines different ordering strategies.
+type SortOrder string
+
+var LegacySortOrder SortOrder = "legacy"
```

Comment:
The linter makes a good point about these (and a couple other variables in test files)--I think they should be constants actually